File: 3784. Minimum Deletion Cost to Make All Characters Equal.java
class Solution {
    public long minCost(String s, int[] cost) {
        HashMap<Character, Long> hm = new HashMap<>();
        long totalCost = 0;
        for(int i = 0; i < s.length(); i++){
            char ch = s.charAt(i);
            hm.put(ch, hm.getOrDefault(ch, 0l) + cost[i]);
            totalCost += cost[i];
        }

        long maxCost = 0;
        for(Map.Entry<Character, Long> ele : hm.entrySet()){
            maxCost = Math.max(maxCost, ele.getValue());
        }

        return totalCost - maxCost;
    }
}

File: 3784. Minimum Deletion Cost to Make All Characters Equal.cpp
class Solution {
public:
    long long minCost(string s, vector<int>& cost) {
        unordered_map<char, long long> hm;
        long long totalCost = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s[i];
            hm[ch] = hm.count(ch) ? hm[ch] + cost[i] : cost[i];
            totalCost += cost[i];
        }

        long long maxCost = 0;
        for (auto& ele : hm) {
            maxCost = max(maxCost, ele.second);
        }

        return totalCost - maxCost;
    }
};
